saleor/discount/tasks.py

```diff
@@ -308,7 +309,7 @@ def update_discounted_prices_task():
     if products_ids:
         products = Product.objects.filter(id__in=products_ids)
         update_discounted_prices_for_promotion(products)
-        update_discounted_prices_task.delay()
+        mark_products_for_recalculate_discounted_price(products_ids)
```

Comment:
I might be wrong but it looks like we will end up in infitie loop. We just processed the `products_ids`, and now we will mark it for recalculate